sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CertificateValidationCheck.cs

```diff
@@ -0,0 +1,299 @@
+        {
+            var locations = ArgumentLocations(c, expression);
+            if (!locations.IsEmpty)
+            {   //Report both, assignemnt as well as all implementation occurances
+                c.Context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, primaryLocation, additionalLocations: locations));
+            }
+        }
+
+        private static bool IsValidationDelegateType(ITypeSymbol type)
+        {
+            if (type.Is(KnownType.System_Net_Security_RemoteCertificateValidationCallback))
+            {
+                return true;
+            }
+            else if (type is INamedTypeSymbol NamedSymbol && type.Is(KnownType.System_Func_T1_T2_T3_T4_TResult))
+            {
+                //HttpClientHandler.ServerCertificateCustomValidationCallback uses Func<HttpRequestMessage, X509Certificate2, X509Chain, SslPolicyErrors, bool>
+                //We're actually looking for Func<Any Sender, X509Certificate2, X509Chain, SslPolicyErrors, bool>
+                var parameters = NamedSymbol.DelegateInvokeMethod.Parameters;
+                return parameters.Length == 4   //And it should! T1, T2, T3, T4
+                    && parameters[0].Type.IsClassOrStruct() //We don't care about common (Object) nor specific (HttpRequestMessage) type of Sender
+                    && parameters[1].Type.Is(KnownType.System_Security_Cryptography_X509Certificates_X509Certificate2)
+                    && parameters[2].Type.Is(KnownType.System_Security_Cryptography_X509Certificates_X509Chain)
+                    && parameters[3].Type.Is(KnownType.System_Net_Security_SslPolicyErrors)
+                    && NamedSymbol.DelegateInvokeMethod.ReturnType.Is(KnownType.System_Boolean);
+            }
+            return false;
+        }
+
+        private static ImmutableArray<Location> ArgumentLocations(InspectionContext c, ExpressionSyntax expression)
+        {
+            var ret = ImmutableArray.CreateBuilder<Location>();
+            switch (expression)
+            {
+                case IdentifierNameSyntax identifier:
+                    var identSymbol = c.Context.SemanticModel.GetSymbolInfo(identifier).Symbol;
+                    if (identSymbol != null && identSymbol.DeclaringSyntaxReferences.Length==1)
+                    {
+                        ret.AddRange(IdentifierLocations(c, identSymbol.DeclaringSyntaxReferences.Single().GetSyntax()));
+                    }
+                    break;
+                case ParenthesizedLambdaExpressionSyntax lambda:
+                    if ((lambda.Body as LiteralExpressionSyntax)?.Kind() == SyntaxKind.TrueLiteralExpression)
+                    {
+                        ret.Add(lambda.Body.GetLocation());   //Code was found guilty for lambda (...) => true
+                    }
+                    else if (lambda.Body is BlockSyntax block)
+                    {
+                        ret.AddRange(BlockLocations(block));
```

Comment:
I just checked the coverage on [SonarCloud](https://sonarcloud.io/code?id=sonaranalyzer-dotnet&line=130&pullRequest=2785&selected=sonaranalyzer-dotnet%3Asrc%2FSonarAnalyzer.CSharp%2FRules%2FCertificateValidationCheck.cs) and this line is actually not covered.
For completeness sake, it would be nice to have one additional test case with lambda having a block body:
```
            CreateRQ().ServerCertificateValidationCallback += (sender, certificate, chain, SslPolicyErrors) =>
            {
                return true;
            };

            CreateRQ().ServerCertificateValidationCallback += (sender, certificate, chain, SslPolicyErrors) =>
            {
                return false;
            };
```